lib/saml2.coffee

```diff
@@ -305,8 +309,7 @@ module.exports.ServiceProvider =
         return cb err if err?
         uri = url.parse identity_provider.sso_login_url
         uri.query =
-          SAMLRequest: deflated.toString 'base64'
-        uri.query.RelayState = relay_state if relay_state?
+          sign_get_request deflated.toString('base64'), @private_key, relay_state
```

Comment:
There's a bit of a coffeescript gotcha here. The `@` in `@private_key` is shorthand for `this` but because it's inside of a callback, `this` won't be what you expect. The solution to this is to change the line 308 from `zlib.deflateRaw xml, (err, deflated) ->` to `zlib.deflateRaw xml, (err, deflated) =>`. The `=>` (fat arrow) at the end will cause the callback to keep the context of the caller and everything will work as expected.